akka-persistence-testkit/src/main/scala/akka/persistence/testkit/internal/TestKitStorage.scala

```diff
@@ -47,15 +51,24 @@ sealed trait InMemStorage[K, R] extends InternalReprSupport[R] {
           Some(value.drop(fromInclusive).take(maxNum))
         else None)
 
+  def removeFirstInExpectNextQueue(key: K): Unit = Option(expectNextQueue.get(key)).foreach(_.poll())
+
+  def firstInExpectNextQueue(key: K): Option[R] = Option(expectNextQueue.get(key)).flatMap { item =>
+    Try(item.element()).toOption.map(toRepr)
+  }
+
   def findOneByIndex(key: K, index: Int): Option[R] =
     Option(eventsMap.get(key))
       .flatMap {
         case (_, value) => if (value.size > index) Some(value(index)) else None
       }
       .map(toRepr)
 
-  def add(key: K, p: R): Unit =
+  def add(key: K, p: R): Unit = {
+    expectNextQueue.putIfAbsent(key, new util.LinkedList[InternalRepr]())
+    expectNextQueue.get(key).add(toInternal(p))
```

Comment:
`LinkedList` isn't thread safe so this could result in unwanted exceptions on concurrent access.
It might be easier to switch over to the `synchronized` approach right away?

Reply:
Fair enough! I will follow up the changes.